### DISCUSSION TITLE
Generate release notes button does not work for release branches

### DISCUSSION THREAD

**Discussion by josienb** [community/community#113181] [Repositories]
### Select Topic Area

Bug

### Body

Hi! In our workflow, we're using backport branches where we cherry pick bugfix commits from `main` onto. (All our relevant PR's are squashed into single commits, with links to the original PR.) We use the "Generate release notes" button in combination with a `.github/release.yml` config to make generating release notes as automated as we can. This works really great on our `main` branch! :heart:

But when we use the button on our `release-major-2024-1` like branches, where all the tags for bugfix releases live, the generated release notes are always empty. The "Full Changelog" link that is included _does_ show the cherry picked commits. Setting the "Previous tag" explicitly to the version before makes no difference either, the generated release notes are still empty.

I expected that this issue would have been raised before, but could not find it here yet. Is this something that we will have to find a way around, or will this be made to work in the future?

**Comment by lin20260407** [comment]
Hello,

Thank you for reaching out and for the detailed explanation of the workflow you're using, as well as the specific challenge you're encountering with generating release notes for your backport branches. It's great to hear that the release notes generation feature works well for your main branch and that you're taking full advantage of the .github/release.yml configuration for automation. ❤️

Regarding the issue with empty release notes on your release-major-2024-1 like branches, this indeed sounds like an unusual behavior, especially considering that the "Full Changelog" link correctly displays the cherry-picked commits. It's possible that this could be related to how the release notes generation feature interprets tags and commits in non-main branches, or how it interacts with cherry-picked commits specifically.

While I don't have an immediate solution, I would recommend a few steps to help us diagnose and potentially address the issue:

Detailed Reproduction Steps: If possible, could you provide a step-by-step guide to reproduce this issue in a public repository? This would greatly help in understanding the exact workflow and identifying any potential gaps.

.github/release.yml Configuration: Sharing a sanitized version of your .github/release.yml configuration might reveal specific settings that could affect the generation process for your backport branches.

Screenshots or Logs: If there are any logs from the release notes generation process or screenshots that show the discrepancy, they could be very helpful.

Previous Discussions: Since you mentioned expecting this issue to have been raised before, have you encountered any similar issues or discussions, even if not exactly the same, that might provide clues?

**Reply by josienb** [reply]
Thanks for your answer! I have created a public repo where I can reproduce the issue here: https://github.com/josienb/repro-release-note-generation. This also contains a sanitized version of our `.github/release.yml` and screenshots in the README of the issue. Please let me know if you need any other info from me for looking into this issue!

**Reply by lin20260407** [reply]
no thank buddy

**Reply by josienb** [reply]
@weitsunglin is there any update on this - will any work be done to solve this by GitHub? If not, we will start thinking of other ways to solve this problem.

**Comment by AlexFernandes-MOVAI** [comment]
I have also seen changes on the way GitHub automatically generates release notes since 2 days , previously an empty "Previous tag" parameter would do a list based on latest pre-release, now it is based on the latest release

**Comment by dmwh-mmonacelli** [comment]
Have there been any updates to this issue?  I experienced the same problem today after trying to generate release notes when the last commit was a cherry-picked commit of a PR into main (into my release branch) - I get the empty release notes.  Is there a workaround until this bug is fixed?

**Comment by TiesWestendorp** [comment]
Any updates on this issue? Would be a great help in our daily work.

**Comment by fmichaut-diff** [comment]
We also deploy "bugfixes" branches after fixes have been merged to main (the bugfix branches contain the previous release commit and a cherry pick of the bugfix), and the Generate Release Notes button skips the cherry-picked commits, resulting in empty release notes.

What's worse is that clicking on the link to show the differences between the 2 releases, we can clearly see the cherry picked commit.
Why is it not included in the release notes ?

**Comment by mapaille** [comment]
Any update on this? I am also facing this issue.

**Comment by tinazhouhui** [comment]
hitting the same wall, from what I have found out, this seems to be by design. Aparetnly, github does not include direct commits in their release notes. it is sort of hinted in their [documentation](https://docs.github.com/en/repositories/releasing-projects-on-github/automatically-generated-release-notes#about-automatically-generated-release-notes):

> Automatically generated release notes include a list of merged pull requests, a list of contributors to the release, and a link to a full changelog.

and I found a relevant discussion [here](https://github.com/orgs/community/discussions/31628).
